packages/runtime/container-runtime/src/orderedClientElection.ts

```diff
@@ -335,11 +342,18 @@ export class OrderedClientElection
     private addClient(client: ILinkedClient, sequenceNumber: number): void {
         if (this.isEligibleFn(client)) {
             this._eligibleCount++;
-            if (this._electedClient === undefined) {
+            if (!this.hasSummarizer) {
                 // Automatically elect latest client
                 this.tryElectingClient(client, sequenceNumber);
             }
         }
+        else if (client.client.details.type === "summmarizer") {
+            // Track this as the summarizer client if there's already an elected client.
+            // If there's no elected client, this must be an on-demand summarizer.
```

Comment:
I do not think this statement is true. I.e. consider this sequence:
- main client starts
- at some point it becomes latest client (maybe immediately on connection).
- It starts summarizer client.
- It gets nacked from connection (because it sends some op, and there are too many unsummarized ops"
- some time passes...
- summarizer container finally boots and connects. It proceeds to "last summary" process.

I think at the end of this sequence, there might be no elected client. So new client booting will ignore summarizer

Reply:
I think the more interesting question:
Does system resolve properly conflict when two clients show up, and each quickly get nacked (due to too many unsumarized ops), but each starts a summarizer container (as each of them was the oldest non-summarized client at some point)? I.e., once both summarizer container boots, the second one should realize first one is still there and exit, right? Or at least not to produce any summarize and wait for first one to exit. This would reduce number of errors in the system, including summarizers stepping on each other.

Reply:
I see your point. I think there's also a problem with a different kind of case, e.g., where an on-demand summarizer client has been created before we have an elected one. I was trying to solve the problem of not letting the on-demand summarizer be treated as the elected one by having _electedClient===undefined mean that we haven't done an election yet, but I don't think that works. I think things become simpler if we can detect that a given summarizer client was created on demand and not automatically. Maybe that could be exposed in ICapabilities.

Reply:
I do have a question about the ordering of "addClient"/"removeClient" messages. If an interactive client spins up a (non-interactive) summarizer client and then gets nacked, is it true that the "addClient" message for the summarizer client will be processed before the "removeClient" message for the interactive client, to reflect the order in which things actually happened? Or is the ordering not clear or guaranteed?

Reply:
No, there is no guarantees here. Non-interactive client takes a while to load and connect, so interactive client may have already disconnected by that time. Today, we allow such summarizer to keep limping along if conditions are right (too many unsummarized ops), even though another summarizer (who is properly elected) could be showing up.

I think the best outcome here - all clients (interactive or not) are taken into account when calculating a summarizer. And if either parent (interactive) or child (non-interactive) client is elected, then corresponding child container keeps going with summaries until it exists (commits or fails to commit last summary).